src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,80 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { actions as currentTodoActions } from '../../features/currentTodo';
 import { Loader } from '../Loader';
+import { useDispatch, useSelector } from 'react-redux';
+import { RootState } from '../../app/store';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
 
 export const TodoModal: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [userData, setUserData] = useState<User | null>(null);
+  const dispatch = useDispatch();
+  const currentTodo = useSelector((state: RootState) => state.currentTodo);
+
+  useEffect(() => {
+    if (currentTodo) {
+      setIsLoading(true);
+
+      getUser(currentTodo?.userId)
+        .then(setUserData)
+        .catch(error => {
+          throw error;
```

Comment:
Throwing an error here without any user feedback might not be the best approach. Consider logging the error or displaying a user-friendly message to inform the user about the issue.